scaling/src/test/java/io/quarkus/ts/openshift/scaling/ScalingOpenShiftIT.java

```diff
@@ -37,92 +45,94 @@ public void scaleBack() {
+    public void scaleDownTest() throws InterruptedException {
+
+        int expectedUpUniqueIdsAmount = EXPECTED_SCALE_UP;
+        givenResourcePath("/scaling");
+        whenScaleTo(EXPECTED_SCALE_UP);
+        thenCheckReplicasAmount(EXPECTED_SCALE_UP);
+        whenMakeRequestTo("/scaling", expectedUpUniqueIdsAmount, OK.getStatusCode(), TIMEOUT_SEC);
+        thenCheckRequestCompleted(true);
+
+        int expectedDownUniqueIdsAmount = EXPECTED_SCALE_DOWN;
+        whenScaleTo(EXPECTED_SCALE_DOWN);
+        thenCheckReplicasAmount(EXPECTED_SCALE_DOWN);
+        whenMakeRequestTo("/scaling", expectedDownUniqueIdsAmount, OK.getStatusCode(), TIMEOUT_SEC);
+        thenCheckRequestCompleted(true);
     }
 
+
     /**
      * Workflow:
      * * Scale down to zero replicas.
      * * Execute an arbitrary minimal sample of requests and verify that all get HTTP 503 response.
      */
     @Test
-    public void scaleToZero() {
-        openShiftUtil.scale(DC_NAME, 0);
+    public void scaleToZero() throws InterruptedException {
+
+        givenResourcePath("/scaling");
+        whenScaleTo(0);
+        thenCheckReplicasAmount(0);
+        IntStream.range(0, 10).forEach(num -> makeHttpScalingRequest("/scaling", SERVICE_UNAVAILABLE.getStatusCode()));
+    }
+
+    private void givenResourcePath(String path) {
+        given()
+                .when().get("/scaling")
+                .then()
+                .statusCode(OK.getStatusCode());
+    }
 
-        assertThat(openShiftUtil.countReadyReplicas(DC_NAME)).isEqualTo(0);
+    private void whenScaleTo(int amount) {
+        openShiftUtil.scale(DC_NAME, amount);
+    }
 
-        for (int i = 0; i < 100; i++) {
-            given()
-                    .when().get("/scaling")
-                    .then()
-                    .statusCode(SERVICE_UNAVAILABLE.getStatusCode());
-        }
+    private void thenCheckReplicasAmount(int expectedAmount) {
+        assertThat(openShiftUtil.countReadyReplicas(DC_NAME)).isEqualTo(expectedAmount);
+    }
+
+    private void thenCheckRequestCompleted(boolean expectedStatus) {
+        assertEquals(expectedStatus, completed, "pods doesn't scale Up/down as expected.");
+    }
+
+    private void whenMakeRequestTo(String path, int requestAmount, int expectedHttpStatus, int timeout) throws InterruptedException {
+        final CountDownLatch latch = new CountDownLatch(requestAmount);
+        Multi<Long> uniqueIds = Multi.createFrom().ticks().every(Duration.ofMillis(DELAY_BETWEEN_REQUEST_MS))
+                .onOverflow().drop()
+                .map(weakUp -> makeHttpScalingRequest(path, expectedHttpStatus))
+                .map(resp -> resp.extract().asString())
+                .map(resp -> Long.parseLong(resp))
+                .transform()
+                .byDroppingDuplicates()
+                .onFailure().recoverWithItem(-1L);
+
+        uniqueIds.subscribe().with(id -> latch.countDown());
+        completed = latch.await(timeout, TimeUnit.SECONDS);
```

Comment:
I think this method is too complicated to follow. If I don't understand it wrongly, what is doing here is to get the distinct timestamps autogenerated by each replica from the GET endpoints. It would be easier to use the Awailability framework:

```java
Set<String> replicas = new HashSet<>();
await().pollInterval(DELAY_BETWEEN_REQUEST_MS, TimeUnit.MILLISECONDS)
          .atMost(TIMEOUT_SEC, TimeUnit.SECONDS)
          .untilAsserted(() -> {
                 replicas.add(makeHttpScalingRequest(path, OK.getStatusCode()).extract().asString());
                 assertEquals(replicas.count(), expectedReplicas);
          });
```

Also, as expectedHttpStatus and timeout arguments are always the same, we could remove them as parameters and use directly the static fields.

Finally, as we're asserting the number of replicas here, we would not need having the completed field and the thenCheckRequestCompleted method.